core/io/config_file.cpp

```diff
@@ -61,80 +61,155 @@ PackedStringArray ConfigFile::_get_section_keys(const String &p_section) const {
 	return arr;
 }
 
+void ConfigFile::_create_entry_if_needed(const String &p_section, const String &p_key) {
+	if (!entries.has(p_section)) {
+		entries.insert(p_section, HashMap<String, Entry>(), p_section.is_empty());
+	}
+	if (!entries[p_section].has(p_key)) {
+		entries[p_section].insert(p_key, Entry());
+	}
+}
+
+String ConfigFile::_get_entry_as_string(const String &p_key, const Entry &p_entry) const {
+	String estr;
+
+	if (!p_entry.comment.is_empty()) {
+		PackedStringArray lines = p_entry.comment.split("\n", true); // allow empty for multi-line separation in detailed comments
```

Comment:
Unsure this comment is necessary, but:
```suggestion
		PackedStringArray lines = p_entry.comment.split("\n", true); // Allow empty for multi-line separation in detailed comments.
```